src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestV4ServiceImpl.java

```diff
@@ -0,0 +1,344 @@
+        ConcurRequestV4PersonDTO owner = requestAsListItem.getOwner();
+        if (ObjectUtils.isNull(owner) || StringUtils.isBlank(owner.getId())) {
+            return false;
+        }
+        boolean isTestUser = testUserIdMappings.containsKey(owner.getId());
+        return isProduction() != isTestUser;
+    }
+
+    protected ConcurEventNotificationProcessingResultsDTO processTravelRequest(String accessToken,
+            ConcurRequestV4ListItemDTO requestAsListItem) {
+        String requestUuid = requestAsListItem.getId();
+        if (!StringUtils.isAlphanumeric(requestUuid)) {
+            throw new RuntimeException("Found a request with an unexpected non-alphanumeric ID: " + requestUuid);
+        }
+        ConcurEventNotificationVersion2ProcessingResults processingResult;
+        List<String> validationMessages = new ArrayList<>();
+        boolean requestValid;
+        
+        try {
+            ConcurRequestV4ReportDTO request = getTravelRequest(accessToken, requestUuid);
+            ConcurAccountInfo accountInfo = buildAccountInfo(request);
+            LOG.info("processTravelRequest, Validating request " + requestUuid + " with account info: "
+                    + accountInfo.toString());
+            ValidationResult validationResult = concurAccountValidationService.validateConcurAccountInfo(accountInfo);
+            requestValid = validationResult.isValid();
+            validationMessages.addAll(validationResult.getMessages());
+            processingResult = requestValid ? ConcurEventNotificationVersion2ProcessingResults.validAccounts
+                    : ConcurEventNotificationVersion2ProcessingResults.invalidAccounts;
+        } catch (Exception e) {
+            LOG.error("processTravelRequest, Unexpected error encountered while validating request " + requestUuid, e);
+            requestValid = false;
+            processingResult = ConcurEventNotificationVersion2ProcessingResults.processingError;
+            validationMessages.add(PROCESSING_ERROR_MESSAGE);
+        }
+        
+        ConcurEventNotificationProcessingResultsDTO resultsDTO = new ConcurEventNotificationProcessingResultsDTO(
+                ConcurEventNoticationVersion2EventType.TravelRequest, processingResult,
+                requestAsListItem.getRequestId(), validationMessages);
+        updateRequestStatusInConcur(accessToken, requestUuid, resultsDTO);
+        
+        return resultsDTO;
+    }
+
+    protected ConcurAccountInfo buildAccountInfo(ConcurRequestV4ReportDTO request) {
+        ConcurAccountInfo accountInfo = new ConcurAccountInfo();
+        accountInfo.setChart(defaultCodeToEmptyIfAbsent(request.getChart()));
+        accountInfo.setAccountNumber(defaultCodeToEmptyIfAbsent(request.getAccount()));
+        accountInfo.setSubAccountNumber(defaultCodeToEmptyIfAbsent(request.getSubAccount()));
+        accountInfo.setObjectCode(concurBatchUtilityService.getConcurParameterValue(
```

Comment:
don't we want to get the object code from the request, and if that is empty, then default to the parameter value

Reply:
Based on the existing code in ConcurReportsServiceImpl.extractAccountInfoFromTravelRequestDetails(), the Travel Request processing always seems to set the object code to the value from the parameter. I would need to double-check with Tami to confirm whether that should continue to remain the case. (Looking at some manually-retrieved travel requests from my earlier testing, there might be an object code in the "custom12" field, but I would need more analysis to confirm that.)

Reply:
Yes, please double check with Tami.  I think the expense report processing looks at the object code of the report, and only uses the default if the report doesn't have an object code.  If we should just use the default object code, the expense report processing may need to be updated.  That can be done on a separate user story.